tests/test_module_state.py

```diff
@@ -0,0 +1,133 @@
+import ast
+import warnings
+import py.path
+import pytest
+import six
+
+from ansible.parsing.metadata import extract_metadata
+
+from .conftest import TEST_PLAYBOOKS
+
+if six.PY2:
+    ast_try = ast.TryExcept
+else:
+    ast_try = ast.Try
+
+SPECIAL_MODULES = {
+    'katello_content_view_version': 'tested as part of katello_content_view',
+}
+
+MODULES_PATH = py.path.local(__file__).realpath() / '..' / '..' / 'plugins' / 'modules'
+
+
+def find_all_modules():
+    for module in MODULES_PATH.listdir(sort=True):
+        module = module.basename.replace('.py', '')
+        if module != '__init__':
+            yield module
+
+
+ALL_MODULES = list(find_all_modules())
+
+
+def _strip_module_prefix(module):
+    return module.replace('foreman_', '').replace('katello_', '')
+
+
+def _module_file_path(module):
+    module_file_name = "{}.py".format(module)
+    return MODULES_PATH / module_file_name
+
+
+def _module_is_tested(module):
+    short_module = _strip_module_prefix(module)
+    module_tested_directly = short_module in TEST_PLAYBOOKS or module in TEST_PLAYBOOKS
+    module_tested_by_another = module in SPECIAL_MODULES
+    return module_tested_directly or module_tested_by_another
+
+
+def _module_framework(module):
+    module_file_path = _module_file_path(module)
+    with module_file_path.open() as module_file:
+        module_ast = ast.parse(module_file.read())
+    return _module_framework_from_body(module_ast.body) or 'unknown'
+
+
+def _module_framework_from_body(body):
+    framework = None
+    for entry in body:
+        if isinstance(entry, ast.ImportFrom):
+            if entry.module == 'ansible.module_utils.ansible_nailgun_cement':
+                framework = 'nailgun'
```

Comment:
Should not happen anymore.

Reply:
you say the check as such can be removed too? yeah, probably right, updated.